### ISSUE TITLE
Debugger crashes the game if 'free()' is used together with [] operator

### ISSUE THREAD

**Issue by hilfazer** [godotengine/godot#20814]
**Godot version:**
3.0.6

**OS/device including version:**

Windows 7 Pro SP1

**Issue description:**

Debugger crashes the game if it goes through line with "free()' function call on Node accessed via [] operator, ie:
`arr[0].free()`

I can't get back to game, is crashes. If i don't use breakpoint everything works fine.
Crash happens for both arrays and dicts. It doesn't happen if [] is not used.

**Steps to reproduce:**
Open Node.gd script and place a breakpoint after a with [] operator and 'free()' call.
Other option is to put breakpoint before 'free()' gets called and let debugger execute it.

**Minimal reproduction project:**

https://github.com/hilfazer/Projects/tree/master/Godot/EngineIssues/DebuggerCrashesOnFree

**Comment by hilfazer** [comment]
Crashes in 3.1.1

**Comment by akien-mga** [comment]
Still reproducible in the current master branch (28599e6). Backtrace:
```
[1] /lib64/libc.so.6(+0x3caf0) [0x7f975e553af0] (??:0)
[2] List<Pair<Variant const*, Variant>, DefaultAllocator>::size() const (/home/akien/Projects/godot/godot.git/./core/list.h:411)
[3] OrderedHashMap<Variant, Variant, VariantHasher, VariantComparator, (unsigned char)3, (unsigned char)8>::size() const (/home/akien/Projects/godot/godot.git/./core/ordered_hash_map.h:275)
[4] Dictionary::empty() const (/home/akien/Projects/godot/godot.git/core/dictionary.cpp:130)
[5] Object::get_property_list(List<PropertyInfo, DefaultAllocator>*, bool) const (/home/akien/Projects/godot/godot.git/core/object.cpp:638)
[6] encode_variant(Variant const&, unsigned char*, int&, bool) (/home/akien/Projects/godot/godot.git/core/io/marshalls.cpp:1084)
[7] encode_variant(Variant const&, unsigned char*, int&, bool) (/home/akien/Projects/godot/godot.git/core/io/marshalls.cpp:1191)
[8] ScriptDebuggerRemote::_put_variable(String const&, Variant const&) (/home/akien/Projects/godot/godot.git/scene/debugger/script_debugger_remote.cpp:114)
[9] ScriptDebuggerRemote::debug(ScriptLanguage*, bool, bool) (/home/akien/Projects/godot/godot.git/scene/debugger/script_debugger_remote.cpp:233)
[10] GDScriptLanguage::debug_break(String const&, bool) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript_editor.cpp:245)
[11] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript_function.cpp:1531 (discriminator 3))
[12] GDScriptInstance::_ml_call_reversed(GDScript*, StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript.cpp:1204)
[13] GDScriptInstance::call_multilevel_reversed(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot.git/modules/gdscript/gdscript.cpp:1213)
[14] Node::_notification(int) (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:152)
[15] Node::_notificationv(int, bool) (/home/akien/Projects/godot/godot.git/./scene/main/node.h:46 (discriminator 14))
[16] Object::notification(int, bool) (/home/akien/Projects/godot/godot.git/core/object.cpp:933)
[17] Node::_propagate_ready() (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:197)
[18] Node::_propagate_ready() (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:186 (discriminator 2))
[19] Node::_set_tree(SceneTree*) (/home/akien/Projects/godot/godot.git/scene/main/node.cpp:2565)
[20] SceneTree::init() (/home/akien/Projects/godot/godot.git/scene/main/scene_tree.cpp:466)
[21] OS_X11::run() (/home/akien/Projects/godot/godot.git/platform/x11/os_x11.cpp:3249)
[22] /home/akien/Projects/godot/godot.git/bin/godot.x11.tools.64(main+0xfa) [0x1423d3c] (/home/akien/Projects/godot/godot.git/platform/x11/godot_x11.cpp:57)
[23] /lib64/libc.so.6(__libc_start_main+0xeb) [0x7f975e53db0b] (??:0)
[24] /home/akien/Projects/godot/godot.git/bin/godot.x11.tools.64(_start+0x2a) [0x1423b9a] (/home/iurt/rpmbuild/BUILD/glibc-2.29/csu/../sysdeps/x86_64/start.S:122)
-- END OF BACKTRACE --
```

**Comment by qarmin** [comment]
This seems to be fixed in Godot 3.2.2
